src/widgets/FormControl.js

```diff
@@ -229,54 +222,102 @@ const FormControlComponent = ({
     [SaveButton, CancelButton]
   );
 
-  const InputConfirm = (
-    <ConfirmForm
-      isOpen={isConfirmFormOpen}
-      questionText={confirmText}
-      onConfirm={onSaveCompletedForm}
-      confirmText={modalStrings.confirm}
-    />
-  );
-
-  const InputDetails = (
-    <View style={localStyles.flexOne}>
-      <ScrollView style={localStyles.whiteBackground}>
-        <View style={localStyles.flexRow}>
-          <View style={localStyles.flexOne} />
-          <View style={localStyles.flexTen}>{formInputs()}</View>
-          <View style={localStyles.flexOne} />
-        </View>
-      </ScrollView>
-      <Buttons />
-    </View>
+  return (
+    <>
+      <View
+        style={
+          isConfirmFormOpen
+            ? { ...localStyles.flexOne, ...localStyles.hidden }
+            : localStyles.flexOne
+        }
+      >
+        <ScrollView style={localStyles.whiteBackground}>
+          <View style={localStyles.flexRow}>
+            <View style={localStyles.flexOne} />
+            <View style={localStyles.flexTen}>{formInputs()}</View>
+            <View style={localStyles.flexOne} />
+          </View>
+        </ScrollView>
+        <Buttons />
+      </View>
+      <View
+        style={
+          isConfirmFormOpen
+            ? localStyles.flexOne
+            : { ...localStyles.flexOne, ...localStyles.hidden }
+        }
+      >
+        <ConfirmForm
+          isOpen={isConfirmFormOpen}
+          questionText={confirmText}
+          onConfirm={onSaveForm}
+          onCancel={onCancelForm}
+          confirmText={modalStrings.confirm}
+        />
+      </View>
+    </>
```

Comment:
A bit of an antipattern here, I think. In order to prevent the form cells from being unmounted (and the refs being nulled, and form state getting all messed up), the form style is set to `display: 'none'`. It's not great at all, but I couldn't get the `ConfirmForm` to hide otherwise (even when `isOpen={false}`, it would still render half the screen).

As it's working, it's probably better to leave as is and open an issue (maybe for `5.1.0`?) if it's worth it!

Reply:
`ConfirmForm` needs to be wrapped in a `ModalContainer` - looks like you were just rendering in within the last modal container which because of flex would take up space

Can make it work doing:

```
    <>
      <ScrollView style={localStyles.whiteBackground}>
        <View style={localStyles.flexRow}>
          <View style={localStyles.flexOne} />
          <View style={localStyles.flexTen}>{formInputs()}</View>
          <View style={localStyles.flexOne} />
        </View>
      </ScrollView>

      <Buttons />

      <ModalContainer fullScreen isVisible={isConfirmFormOpen} noCancel>
        <ConfirmForm
          questionText={confirmText}
          onConfirm={onSaveForm}
          onCancel={onCancelForm}
          confirmText={modalStrings.confirm}
        />
      </ModalContainer>
    </>
```

just a couple lines, probs worth it

Reply:
Ah yup, makes sense. Maybe `ConfirmForm` should be called `ConfirmModal` or something similar, just to make that a bit clearer?

Reply:
https://github.com/openmsupply/mobile/pull/1943#discussion_r379949015

🤷

Reply:
:smiley: maybe something for a new issue as it looks from that checklist like there is some migration remaining to do also?

Reply:
yeah heh